src/moonlink/src/row/column_array_builder.rs

```diff
@@ -322,6 +405,54 @@ impl ColumnArrayBuilder {
                 };
                 Ok(())
             }
+            ColumnArrayBuilder::Struct(builder, array_helper) => {
+                match value {
+                    RowValue::Struct(fields) => {
+                        for (i, field) in fields.iter().enumerate().take(builder.num_fields()) {
+                            Self::append_value_to_struct_field(builder, i, field)?;
+                        }
+                        builder.append(true);
+                    }
+                    RowValue::Array(v) => {
+                        array_helper.as_mut().unwrap().push(builder.len() as i32);
+                        for item in v {
+                            match item {
+                                RowValue::Struct(fields) => {
+                                    for (i, field) in
+                                        fields.iter().enumerate().take(builder.num_fields())
+                                    {
+                                        Self::append_value_to_struct_field(builder, i, field)?;
+                                    }
+                                    builder.append(true);
+                                }
+                                RowValue::Null => {
+                                    for i in 0..builder.num_fields() {
+                                        Self::append_value_to_struct_field(
+                                            builder,
+                                            i,
+                                            &RowValue::Null,
+                                        )?;
+                                    }
+                                    builder.append(false);
+                                }
+                                _ => unreachable!("Struct expected from well-typed input"),
+                            }
+                        }
+                    }
+                    RowValue::Null => {
+                        if let Some(helper) = array_helper.as_mut() {
+                            helper.push_null();
+                        } else {
+                            for i in 0..builder.num_fields() {
+                                Self::append_value_to_struct_field(builder, i, &RowValue::Null)?;
```

Comment:
I'm not sure this is the correct implementation, my understanding for adding NULL-field wise is:
suppose you have 3 primitive fields in the struct, field-1 and field-2 are not null, field-3 is null
so you need to fill in field-3-builder with null, and mark the whole struct append as valid

But here, it's clear the struct itself is NULL, I think we should simply declare it null with
```rs
builder.append(null);
```

Reply:
In one word, I think 
(1) declare struct field as null
vs
(2) declare the whole struct as null
is different, and we should implement (2) here